scene/main/http_request.cpp

```diff
@@ -207,6 +208,22 @@ void HTTPRequest::cancel_request() {
 	requesting = false;
 }
 
+bool HTTPRequest::_is_automatic_redirect() {
+	if (unlikely(method == HTTPClient::METHOD_CONNECT)) {
+		// Never automatically redirect CONNECT requests.
+		return false;
+	} else if (response_code == 301 || response_code == 302 || response_code == 303 || response_code == 305) {
+		// We change unsafe methods to GET for 301, 302, and 303, so these are always redirected.
+		// 305 is deprecated and treated as equivalent to 302.
+		return true;
+	} else if ((response_code == 307 || response_code == 308) && (method == HTTPClient::METHOD_GET || method == HTTPClient::METHOD_HEAD || method == HTTPClient::METHOD_OPTIONS || method == HTTPClient::METHOD_TRACE)) {
```

Comment:
Extract this into a function (or multiple functions). Way too unwieldy.

Reply:
Thank you for your help on this in RC. I have moved the check for the safe methods into a function `_is_method_safe`. This is the same method that was added in #91199, and both PRs modify the same lines, so there will be a conflict once either PR is merged, but it's simple to resolve, and I should get an email once one of these PRs is merged, so I can take care of it right away.

I chose to leave the conditionals on the actual HTTP status codes in the `_is_automatic_redirect` method unchanged because I think it makes it explicit that we are doing automatic redirects for specific status codes. I considered adding a function `_is_redirect_method_preserving`, which would check for status codes 307 and 308, but I felt like it was unnecessary from a readability perspective.

Please let me know if you have any questions for me. Thanks!